frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BasePushBranchAction.java

```diff
@@ -57,7 +63,29 @@ public IEnhancedLambdaLogger log() {
   @UIEffect
   public void onUpdate(AnActionEvent anActionEvent) {
     super.onUpdate(anActionEvent);
+
     syncToRemoteStatusDependentActionUpdate(anActionEvent);
+
+    var branchName = getNameOfBranchUnderActionWithLogging(anActionEvent);
+    var relation = branchName.flatMap(bn -> getGitMacheteBranchByNameWithLogging(anActionEvent, bn))
+        .map(b -> b.getSyncToRemoteStatus())
+        .map(strs -> strs.getRelation());
+    var project = tryGetProject(anActionEvent);
```

Comment:
Hmm given that we're here in a class that extends `BaseProjectKeyAvailabilityAssuranceAction`... I don't think that using `tryGetProject` (instead of `getProject` makes sense); at this point we're pretty much sure that `PROJECT` key is available. Also, see #533